src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,82 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { Loader } from '../Loader';
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, addComment, removeComment } from '../../api/comments';
+import { CommentsDetails } from '../CommentsDetails';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ postId }) => {
+  const [comments, setComments] = useState([]);
+  const [post, setPost] = useState(null);
+
+  const add = async(newComment) => {
+    await addComment(newComment);
+
+    setComments([...comments, newComment]);
+  };
+
+  const remove = async(removedCommentId) => {
+    await removeComment(removedCommentId);
+    const filteredComments = comments.filter(
+      comment => removedCommentId !== comment.id,
+    );
+
+    setComments(filteredComments);
+  };
+
+  useEffect(() => {
+    const getDetails = async() => {
+      const postFromServer = await getPostDetails(postId);
+
+      setPost(postFromServer);
+    };
+
+    const getComments = async() => {
+      const commentsFromServer = await getPostComments(postId);
+
+      setComments(commentsFromServer);
+    };
+
+    getDetails();
+    getComments();
+  }, [postId]);
+
+  return (
+    <>
+      {!post ? (
+        <Loader />
+      ) : (
+        <div className="PostDetails">
```

Comment:
еще можно сделать так
```suggestion
  if (!post) {
    return <Loader />;  
  }
  
  return (
        <div className="PostDetails">
```